x-pack/elastic-agent/pkg/agent/application/handler_action_policy_change.go

```diff
@@ -31,9 +50,81 @@ func (h *handlerPolicyChange) Handle(ctx context.Context, a action, acker fleetA
 	}
 
 	h.log.Debugf("handlerPolicyChange: emit configuration for action %+v", a)
+	err = h.handleKibanaHosts(c)
+	if err != nil {
+		return err
+	}
 	if err := h.emitter(c); err != nil {
 		return err
 	}
 
 	return acker.Ack(ctx, action)
 }
+
+func (h *handlerPolicyChange) handleKibanaHosts(c *config.Config) error {
+	cfg, err := configuration.NewFromConfig(c)
+	if err != nil {
+		return errors.New(err, "could not parse the configuration from the policy", errors.TypeConfig)
+	}
+	if kibanaEqual(h.config.Fleet.Kibana, cfg.Fleet.Kibana) {
+		// already the same hosts
+		return nil
+	}
+	// only set protocol/hosts as that is all Fleet currently sends
+	h.config.Fleet.Kibana.Protocol = cfg.Fleet.Kibana.Protocol
```

Comment:
in case client creation fails we will end up with updated values in memory which does not correspond to currently running client, we should probably do some rollback on error

Reply:
Good call, I have added in rollback of the previous values on failure.